python/anvill/binja.py

```diff
@@ -173,199 +173,251 @@ def _convert_bn_llil_type(
+            return self.__convert_typedef(named_tinfo)
 
-def get_type(bv, ty):
-    """Type class that gives access to type sizes, printings, etc."""
+        elif named_tinfo.type_class == bn.NamedTypeReferenceClass.EnumNamedTypeClass:
+            return self.__convert_enum(ref_type)
 
-    if isinstance(ty, Type):
-        return ty
+        else:
+            DEBUG("WARNING: Unknown named type {} not handled".format(named_tinfo))
 
-    elif isinstance(ty, Function):
-        return ty.type()
+    def _convert_bn_type(self, tinfo: bn.types.Type):
+        """Convert an bn `Type` instance into a `Type` instance."""
 
-    elif isinstance(ty, bn.Type):
-        return _convert_bn_type(bv, ty, {})
+        if _cache_key(tinfo) in self._cache:
+            return self._cache[_cache_key(tinfo)]
 
-    if not ty:
-        return VoidType()
+        # Void type
+        if tinfo.type_class == bn.TypeClass.VoidTypeClass:
+            return VoidType()
 
-    raise UnhandledTypeException("Unrecognized type passed to `Type`.", ty)
+        elif tinfo.type_class == bn.TypeClass.PointerTypeClass:
+            return self.__convert_pointer(tinfo)
 
+        elif tinfo.type_class == bn.TypeClass.FunctionTypeClass:
+            return self.__convert_function(tinfo)
 
-def get_arch(bv):
-    """Arch class that gives access to architecture-specific functionality."""
-    name = bv.arch.name
-    if name == "x86_64":
-        return AMD64Arch()
-    elif name == "x86":
-        return X86Arch()
-    elif name == "aarch64":
-        return AArch64Arch()
-    else:
-        raise UnhandledArchitectureType(
-            "Missing architecture object type for architecture '{}'".format(name)
-        )
+        elif tinfo.type_class == bn.TypeClass.ArrayTypeClass:
+            return self.__convert_array(tinfo)
 
+        elif tinfo.type_class == bn.TypeClass.StructureTypeClass:
+            return self.__convert_struct(tinfo)
 
-def get_os(bv):
-    """OS class that gives access to OS-specific functionality."""
-    platform = str(bv.platform)
-    if "linux" in platform:
-        return LinuxOS()
-    elif "mac" in platform:
-        return MacOS()
-    elif "windows" in platform:
-        return WindowsOS()
-    else:
-        raise UnhandledOSException(
-            "Missing operating system object type for OS '{}'".format(platform)
-        )
+        elif tinfo.type_class == bn.TypeClass.EnumerationTypeClass:
+            return self.__convert_enum(tinfo)
+
+        elif tinfo.type_class == bn.TypeClass.BoolTypeClass:
+            return BoolType()
+
+        elif tinfo.type_class == bn.TypeClass.IntegerTypeClass:
+            return self.__convert_integer(tinfo)
+
+        elif tinfo.type_class == bn.TypeClass.FloatTypeClass:
+            return FloatingPointType(tinfo.width)
+
+        elif tinfo.type_class == bn.TypeClass.NamedTypeReferenceClass:
+            return self.__convert_named_reference(tinfo)
+
+        elif tinfo.type_class in [
```

Comment:
Make the `err_type_class` dictionary a global var, then check if `tinfo.type_class` is in the dict, i.e. is a key in the dict.